src/cmd/ksh93/tests/emacs.exp

```diff
@@ -0,0 +1,279 @@
+# Check emacs keybindings
+set pid [spawn $ksh]
+expect_prompt
+
+send "echo hello\r"
+expect "\r\nhello\r\n" {
+    puts "echo works"
+}
+
+expect_prompt
+
+# Enable emacs keybindings
+send "set -o emacs\r"
+expect_prompt
+
+# Send ctrl-p to check previous command
+send "\x10"
+expect -exact "set -o emacs" {
+    puts "ctrl-p works"
+}
+
+send "\x10"
+expect -exact "echo hello"
+
+# Send ctrl-n to check next command
+send "\x0E"
+expect -exact "set -o emacs" {
+    puts "ctrl-n works"
+}
+
+# Take cursor at beginning of line with ctrl-a
+send "\x01"
+expect -exact "" {
```

Comment:
This is always true regardless of what operation was just done. The only way to test that [ctrl-a] works is to modify the buffer (e.g., by inserting a new char at the start of the line) then verifying the line looks like we expect.